File: reversenum_leet.py
#Practice
# print(len(y))
#print(y)
# y=y.replace("-","")
#print(y[0])
# l=list(y)
# l[0]=""
# y="".join(l)
# print('y: ',y)
# z=''.join(reversed(y))
# print(z)

# Method 1:
x=-1234
print('Method 1')
print('Before: ', x)
y=str(x)
if y[0]=='-' :
    flag=1
y=y.replace("-","")
z=''.join(reversed(y))
if flag==1:
    z='-'+z;
print(' After: ',z)
print('\n')

#Method 2:
x=-1234
print('Method 2')
print('Before: ', x)
y=str(x)
if y[0]=='-' :
    flag=1
l=list(y)
l[0]=""
y="".join(l)
z=''.join(reversed(y))
if flag==1:
    z='-'+z;
print(' After: ',z)
print('\n')

#Method 3:
x=-1234
print('Method 3')
print('Before: ', x)
y=str(x)
if y[0]=='-' :
    flag=1
l=list(y)
del(l[0])
y="".join(l)
z=''.join(reversed(y))
if flag==1:
    z='-'+z;
print(' After: ',z)
print('\n')

#Method 4
x=-1234
print('Method 4')
print('Before: ', x)
y=list(l)
if y[0]=='-' :
    flag=1
    del(l[0])
if len(y)%2==0 :
    count=len(y)/2
else:
    count=(len(y)/2)-1

for i in range(0,int(count)):
    temp=y[i]
    y[i]=y[len(y)-i-1];
    y[len(y)-i-1]=temp;
z="".join(y)
if flag==1:
    z='-'+z;
print(' After: ',z)
print('\n')

File: Version1/reversenum_leet.cpp
#include<iostream>
using namespace std;
#include<cstdlib>
#include<cstring>
#include<stdio.h>


int main()
{
long x=-12;
string y=to_string(x);
cout<<"Before: "<<y<<endl;
int flag=0;
if(y[0]=='-')
{
  y.erase(0,1);
  flag=1;
}

//Logic 1:
int itr=0;
        int count=y.length();
        if(count==2)
        {
            itr=1;
        }
        else
        {
          itr=count/2;
        }
        for(int i=0; i<itr; i++)
        {
          long temp;
          temp=y[i];
          y[i]=y[(count-1)-i];
          y[(count-1)-i]=temp;
        }

//Logic 2:

//reverse(y.begin(), y.end());
try
{
  signed int new1=stoi(y);
  if(flag==1)
  {
    y="-"+y;
    new1=(0-new1);
  }

  cout<<"AFTER: "<<new1<<endl;

}
catch(const std::exception& e)
{
  cout<<"return 0"<<endl;
}

}
